frontend/src/metabase/visualizations/visualizations/Sankey.jsx

```diff
@@ -0,0 +1,234 @@
+/* eslint-disable react/prop-types */
+import React, { Component } from "react";
+import ReactDOM from "react-dom";
+import { t } from "ttag";
+import d3 from "d3";
+import cx from "classnames";
+
+import _ from "underscore";
+
+import { isNumeric } from "metabase-lib/types/utils/isa";
+
+import { sankey, sankeyLinkHorizontal } from "d3-sankey";
+import chroma from "chroma-js";
+
+import {
+  SANKEY_DATA_SETTINGS,
+} from "../lib/settings/graph";
+
+const MAX_WIDTH = 500;
+const PADDING_BOTTOM = 10;
+
+export default class Sankey extends Component {
+  static uiName = t`Sankey`;
+  static identifier = "sankey";
+  static iconName = "sankey";
+
+  static minSize = { width: 4, height: 4 };
+
+  static isSensible({ cols, rows }) {
+    return rows.length >= 1 && cols.length >= 3;
+  }
+
+  static checkRenderable([
+    {
+      data: { cols, rows },
+    },
+  ]) {
+    if (!isNumeric(cols[0])) {
+      throw new Error(t`Sankey visualization requires a number.`);
+    }
+  }
+
+  state = {
+    mounted: false,
+  };
+
+  _label;
+
+  static settings = {
+    ...SANKEY_DATA_SETTINGS,
+  };
+
+  componentDidMount() {
+    this.setState({ mounted: true });
+  }
+  componentDidUpdate() {
+  }
+
+  getFilteredData(cols, rows, start, destinations, metrics){
+    var dimensions = start.concat(destinations);
```

Comment:
opt for `const` and `let` most of the time instead of `var`

Reply:
I think all of these are fixed.